modin/experimental/engines/omnisci_on_native/io.py

```diff
@@ -344,6 +275,9 @@ def read_csv(
                 null_values=None,
                 true_values=None,
                 false_values=None,
+                # timestamp fields should be handled as strings if parse_dates
+                # didn't passed explicitly as an array or a dict
+                timestamp_parsers=[""] if isinstance(parse_dates, bool) else None,
```

Comment:
Why we don't care what `parse_dates` value is if it's bool? I mean, don't `True` and `False` have different meaning here?

Reply:
it seems that `parse_dates` is handled incorrectly even in the current master...

anyway, this commit just reverts `read_csv` to the state it's presented in the current master. Do we want to do any changes that are not directly related to the actual issue here (#3216)?

Reply:
No, let's keep unrelated changes out. If that's a simple revert let it be so.